drivers/sensors/ltr308.c

```diff
@@ -0,0 +1,1000 @@
+ *
+ ****************************************************************************/
+
+static int ltr308_get_intrmode(FAR struct ltr308_dev_s *priv,
+                               FAR bool *on)
+{
+  uint8_t interrupt_control;
+  int ret;
+
+  ret = ltr308_get_reg8(priv, LTR308_INTERRUPT, &interrupt_control);
+  if (ret < 0)
+    {
+      return ret;
+    }
+
+  *on = interrupt_control & 0x04;
+
+  return ret;
+}
+
+/****************************************************************************
+ * Name: ltr308_set_intrpers
+ *
+ * Description:
+ *   Set up interrupt persistance (controls the N number of times the
+ *   measurement data is outside the range defined by upper and lower
+ *   threshold)
+ *
+ ****************************************************************************/
+
+static int ltr308_set_intrpers(FAR struct ltr308_dev_s *priv,
+                               uint8_t intr_pers)
+{
+  /* Sanity check */
+
+  if (intr_pers >= 16)
+    {
+      intr_pers = 0;
+    }
+
+  return ltr308_set_reg8(priv, LTR308_INTR_PERS, intr_pers << 4);
+}
+
+/****************************************************************************
+ * Name: ltr308_get_intrpers
+ *
+ ****************************************************************************/
+
+static int ltr308_get_intrpers(FAR struct ltr308_dev_s *priv,
+                               FAR uint8_t *intr_pers)
+{
+  int ret;
+
+  ret = ltr308_get_reg8(priv, LTR308_INTR_PERS, intr_pers);
+  if (ret < 0)
+    {
+      return ret;
+    }
+
+  *intr_pers >>= 4;
+
+  return ret;
+}
+
+/****************************************************************************
+ * Name: ltr308_set_threshold
+ *
+ * Description:
+ *   Set the upper and lower limits of the threshold
+ *
+ ****************************************************************************/
+
+static inline int ltr308_set_threshold(FAR struct ltr308_dev_s *priv,
+                                       uint32_t upper_limit,
+                                       uint32_t lower_limit)
+{
+  return ltr308_set_reg24(priv, LTR308_THRES_UP_0, upper_limit) |
+         ltr308_set_reg24(priv, LTR308_THRES_LOW_0, lower_limit);
+}
+
+/****************************************************************************
+ * Name: ltr308_get_threshold
+ *
+ ****************************************************************************/
+
+static inline int ltr308_get_threshold(FAR struct ltr308_dev_s *priv,
+                                       FAR uint32_t *upper_limit,
+                                       FAR uint32_t *lower_limit)
+{
+  return ltr308_get_reg24(priv, LTR308_THRES_UP_0, upper_limit) |
+         ltr308_get_reg24(priv, LTR308_THRES_LOW_0, lower_limit);
+}
+
+/****************************************************************************
+ * Name: ltr308_get_lux
+ *   Convert raw data to lux
+ *
+ ****************************************************************************/
+
+static int ltr308_get_lux(FAR struct ltr308_dev_s *priv, uint8_t gain,
```

Comment:
Maybe can be void?

Reply:
Hmm, I think we won't be able to to signal -ENODATA back to the user if the function is void. And we need to do so if the sensor in saturated